File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/features/auth/dto/UserProfileResponse.java
package edu.cit.garbo.pawnscan.features.auth.dto;

import edu.cit.garbo.pawnscan.features.businessprofile.dto.BusinessProfileResponse;
import edu.cit.garbo.pawnscan.shared.user.RegistrationStatus;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserProfileResponse {

    private Long userId;
    private String email;
    private String fullName;
    private String phoneNumber;
    private String role;
    private RegistrationStatus registrationStatus;
    private LocalDateTime createdAt;
    private BusinessProfileResponse businessProfile;
    private String message;
}

File: mobile/app/src/main/java/com/cit/pawnscan/features/auth/api/UserProfileResponse.kt
package com.cit.pawnscan.features.auth.api

data class UserProfileResponse(
    val userId: Long?,
    val email: String?,
    val fullName: String?,
    val phoneNumber: String?,
    val role: String?,
    val registrationStatus: String?,
    val createdAt: String?,
    val businessProfile: BusinessProfileResponse?,
    val message: String?
)
